packages/amplify-codegen/src/commands/statements.js

```diff
@@ -50,45 +50,52 @@ async function generateStatements(context, forceDownloadSchema, maxDepth, withou
       frontend = decoupleFrontend;
     }
     const language = frontend === 'javascript' ? cfg.amplifyExtension.codeGenTarget : 'graphql';
+    if (!Object.keys(FILE_EXTENSION_MAP).includes(language)) {
+      throw new Error(`Language ${language} not supported`);
+    }
+
     const opsGenSpinner = new Ora(constants.INFO_MESSAGE_OPS_GEN);
     opsGenSpinner.start();
 
     try {
       fs.ensureDirSync(opsGenDirectory);
-      const schemaData = loadSchema(schemaPath);
+      const schemaData = readSchemaFromFile(schemaPath);
       const generatedOps = generateGraphQLDocuments(schemaData, {
-        maxDepth: maxDepth || cfg.amplifyExtension.maxDepth
+        maxDepth: maxDepth || cfg.amplifyExtension.maxDepth,
+        useExternalFragmentForS3Object: (language === 'graphql')
       });
-      await writeGeneratedStatements(language, generatedOps, opsGenDirectory, true);
-      opsGenSpinner.succeed(constants.INFO_MESSAGE_OPS_GEN_SUCCESS + path.relative(path.resolve('.'), opsGenDirectory));
+      if(!generatedOps) {
+        context.print.warning('No GraphQL statements are generated. Check if the introspection schema has GraphQL operations defined.');
+      }
+      else {
+        await writeGeneratedDocuments(language, generatedOps, opsGenDirectory);
+        opsGenSpinner.succeed(constants.INFO_MESSAGE_OPS_GEN_SUCCESS + path.relative(path.resolve('.'), opsGenDirectory));
+      }
     } finally {
       opsGenSpinner.stop();
     }
   }
 }
 
-async function writeGeneratedStatements(language, generatedStatements, outputPath, separateFiles) {
+async function writeGeneratedDocuments(language, generatedStatements, outputPath) {
   const fileExtension = FILE_EXTENSION_MAP[language];
-  if(!generatedStatements) {
-    return;
-  }
-  if (separateFiles) {
-    ['queries', 'mutations', 'subscriptions', 'fragments'].forEach(op => {
-      const ops = generatedStatements[op];
-      if (ops.size) {
-        // This will change post top level codegen changes
-        fs.writeFileSync(path.resolve(path.join(outputPath, `${op}.${fileExtension}`)), JSON.stringify(Object.fromEntries(ops)));
-      }
-    });
-  } else {
-    const ops = [
-      ...generatedStatements.queries,
-      ...generatedStatements.mutations,
-      ...generatedStatements.subscriptions,
-      ...generatedStatements.fragments
-    ].join();
-    if (ops.length) {
-      fs.writeFileSync(path.resolve(outputPath), ops);
+
+  ['queries', 'mutations', 'subscriptions'].forEach(op => {
+    const ops = generatedStatements[op];
+    if (ops && ops.size) {
+      const formattedStatements = (new GraphQLStatementsFormatter(language)).format(ops);
+      const outputFile = path.resolve(path.join(outputPath, `${op}.${fileExtension}`));
+      fs.writeFileSync(outputFile, formattedStatements);
+    }
+  });
+
+  if (fileExtension === 'graphql') {
+    // External Fragments are rendered only for GraphQL targets
```

Comment:
Is this comment originated from the doc-gen implementation? Do we prevent fragments from writing to file system when they are  generated under non-GQL language? If not, I suggest the fragments part to be merged to the code block above as it will check the `ops && ops.size` anyway and use the same format class/methods

Reply:
There was no straight forward comment, but if you look at the template for non-graphql language, say, JS for example, you can see that it [uses `renderToVariable` partial](https://github.com/aws-amplify/amplify-codegen/blob/main/packages/graphql-docs-generator/templates/javascript.hbs) which in turn looks like this:
```
{{#each operations }}
  export const {{formatName name}} =  /* GraphQL */`{{#format }}
      {{> renderOp }}
    {{/format}}`;
{{/each}}
```
But for GraphQL, the template looks like:
```
# this is an auto generated file. This will be overwritten
{{#each operations }}
  {{> renderOp }}
{{/each}}
{{#each fragments }}
  {{> renderExternalFragment }}
{{/each}}
```
So for non-graphql languages, we're ignoring the fragments arg that gets passed to the [render method](https://github.com/aws-amplify/amplify-codegen/blob/066615e97c54e9f62b599991e058fc0d70aa22b4/packages/graphql-docs-generator/src/index.ts#L62)